### PR TITLE
Disable MongoDB server selection in tests

### LINKED ISSUE #13953: `TestTLSConfiguration/insecure_ignores_incorrect_CN/mongod` flakiness
#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds/9bddb91b-64d8-4af5-9364-1b43aa68434f?project=ci-account

#### Relevant snippet

```
        	Test:       	TestTLSConfiguration/insecure_ignores_incorrect_CN/mongodb
        	            	server selection error: server selection timeout, current topology: { Type: Unknown, Servers: [{ Addr: 127.0.0.1:36507, Type: Unknown, Average RTT: 0 }, ] }
        	Error:      	Received unexpected error:
        	Error Trace:	ca_test.go:467
    ca_test.go:467: 
```

### PR DESCRIPTION
**Pull request by jakule**
Why?
----

MongoDB server selection logic in tests causes many problems as the mongo driver retires connection multiple times, often hitting a test timeout and making our tests flaky. 

Solution
---

The newer version of the mongo driver has a new option "Load Balancer" that allows forcing the driver to stop re-trying and just return the first error returned by the server.

Note
---
This PR also updates the mongo driver to v1.10.0.

Closes #13953

### CODE REVIEW

File: go.mod
Comment (marcoandredinis) [inline go.mod:51]: I don't think we should remove `license` and `reporting`
This usually happens when you don't have the `e/` submodule checked out

Comment (jakule) [reply go.mod:51]: You're right. Fixed

Comment (marcoandredinis) [reply go.mod:51]: Sorry, I'm still seeing the files being removed
Did you push? :see_no_evil:

**Review by smallinsky — APPROVED** [review]

File: go.mod
Comment (zmb3) [inline go.mod:51]: Don't forget to revert this.

**Review by zmb3 — APPROVED** [review]
Looks good to me so long as Marco's comment about the `e` dependencies is fixed.

File: go.mod
Comment (jakule) [reply go.mod]: I reverted this change and then forgot to push it 🤦 Fixed. Thanks.

**Review by r0mant — APPROVED** [review]
lgtm but if we want to backport this to branch/v10, let's do some Mongo testing to make sure driver upgrade didn't break anything.

cc @smallinsky

**Comment by jakule** [comment]
> lgtm but if we want to backport this to branch/v10, let's do some Mongo testing to make sure driver upgrade didn't break anything.
> 
> cc @smallinsky

I assumed that we don't want to backport it immediately. I'll wait for the green light before doing that.

**Pull request merged**
